uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -87,12 +88,13 @@ use super::{APIConverter, ComponentInterface};
 ///
 /// Enums are passed across the FFI by serializing to a bytebuffer, with a
 /// i32 indicating the variant followed by the serialization of each field.
-#[derive(Debug, Clone, Hash)]
+#[derive(Debug, Clone, Hash, Default)]
 pub struct Enum {
     pub(super) name: String,
     pub(super) variants: Vec<Variant>,
     // "Flat" enums do not have, and will never have, variants with associated data.
     pub(super) flat: bool,
+    pub(super) docs: Vec<String>,
```

Comment:
One neat thing that we get pretty much for free from parsing Rust code, is *excellent* support for inline docs on all items. Here, I've given each `ComponentInterface` item a simple `docs` attribute to let it hold any docs parsed from the Rust code, and have spat the docs back out again when converting it into a `.udl`. file. You can imagine a future where we carry the docs through to the generated Kotlin, Swift etc files as well.

Reply:
The fact the ffi is bound much tighter to the actual impl really is the key feature here, and doc generation is just the most obvious manifestation of that.